Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
-#include "../include/Volunteer.h";
+
+#include "../include/Volunteer.h"
+#include "../include/Order.h"
+
+
+/////////////////////////////////////////////////////////
+//////////////////////// Volunteer //////////////////////
+/////////////////////////////////////////////////////////
+
+Volunteer::Volunteer(int id, const string &name) : id(id), name(name) 
+{
+    completedOrderId = -1;
+    activeOrderId = -1;
+}
+
+Volunteer::Volunteer(int id, const string &name, int completedOrderId, int activeOrderId) : id(id), name(name), completedOrderId(completedOrderId), activeOrderId(activeOrderId) {}
+
+int Volunteer::getId() const
+{
+    return id;
+}
+
+const string& Volunteer::getName() const 
+{
+    return name;
+}
+
+int Volunteer::getActiveOrderId() const 
+{
+    return activeOrderId;
+}
+
+bool Volunteer::isBusy() const
+{
+    return (activeOrderId != NO_ORDER);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// Collector Volunteer ////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+CollectorVolunteer::CollectorVolunteer(int id, const string &name, const int coolDown) : Volunteer(id, name) ,coolDown(coolDown), timeLeft(NO_ORDER){}
+CollectorVolunteer* CollectorVolunteer::clone() const 
+{
+     return new CollectorVolunteer(*this);
+}
+void CollectorVolunteer::step() // need to check if it's correct
```

Comment:
add foe every step implementation condition in the begining:
canTakeOrder(), i prefer to give the step responsibility to each class, if cant take order just finish the function